pkg/vgmanager/cleanup.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vgmanager
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/red-hat-storage/lvm-operator/controllers"
+	topolvmv1 "github.com/topolvm/topolvm/api/v1"
+	lvmCMD "github.com/topolvm/topolvm/pkg/lvmd/cmd"
+	crc "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+// vgCleanup removes all volume groups existing in lvmd config file and on every vg
+// removal it also removed config file
+func vgCleanup(r *VGReconciler, ctx context.Context) error {
+
+	// get limited number of logical volume CRs to verify their presence
+	lvs := &topolvmv1.LogicalVolumeList{}
+	if err := r.Client.List(ctx, lvs, &crc.ListOptions{Limit: int64(1)}); err == nil {
+		// do not proceed with cleanup if there are logicalVolumes
+		if len(lvs.Items) > 0 {
+			return fmt.Errorf("found presence of logicalvolumes and aborting volume group cleanup")
+		}
+	} else {
+		return err
+	}
+
+	// get all the volume groups from lvmd file
+	lvmdConfig := &lvmCMD.Config{
```

Comment:
This should be a global var in the package or an argument to the function. Or an arg in the reconciler.

Reply:
why should this be the case? my rationale is, if we fail to delete any VG then we need to read remaining content from config file in next reconcile and then this struct has to be re-declared again.

Reply:
This struct should be the same one created by the VGmanager when creating the file. Don't regenerate it each time.

Reply:
Actually scratch the last comment. What we should be doing is saving the socket and filename that was used when the file was created and using those in the struct.